src/avalanchemq/queue/queue.cr

```diff
@@ -536,19 +536,18 @@ module AvalancheMQ
         if expire_at = expire_at(sp)
           expire_in = expire_at - now
           @log.debug { "expire sp=#{sp} expire_in=#{expire_in}" }
-          if expire_in > 0
-            @log.debug { "No more message to expire" }
-            break
-          end
-
-          expire_msg(sp, :expired)
-          if (i += 1) == 8192
-            Fiber.yield
-            i = 0
+          if expire_in < 0
+            expire_msg(sp, :expired)
+            if (i += 1) == 8192
+              Fiber.yield
+              i = 0
+            end
+            true
+          else
+            false
           end
-          true
         else
-          break
+          false
```

Comment:
we won't break early now? so all messages in the ready queue will always be tested? or does returning false break?

Reply:
`shift`  method in ReadyQueue breaks if block returns false